src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.Windows.cs

```diff
@@ -36,6 +42,202 @@ internal IntPtr[] GetWaitedHandleArray(int requiredCapacity)
+            _priority = MapFromOSPriority(Interop.mincore.GetThreadPriority(_osHandle));
+        }
+
+        private static SafeWaitHandle GetOSHandleForCurrentThread()
+        {
+            IntPtr currentProcHandle = Interop.mincore.GetCurrentProcess();
+            IntPtr currentThreadHandle = Interop.mincore.GetCurrentThread();
+            SafeWaitHandle threadHandle;
+
+            if (Interop.mincore.DuplicateHandle(currentProcHandle, currentThreadHandle, currentProcHandle,
+                out threadHandle, 0, false, (uint)Interop.Constants.DuplicateSameAccess))
+            {
+                return threadHandle;
+            }
+
+            // Throw an ApplicationException for compatibility with CoreCLR. First save the error code.
+            int errorCode = Marshal.GetLastWin32Error();
+            var ex = new ApplicationException();
+            ex.SetErrorCode(errorCode);
+            throw ex;
+        }
+
+        private static ThreadPriority MapFromOSPriority(OSThreadPriority priority)
+        {
+            switch (priority)
+            {
+                case OSThreadPriority.Idle:
+                case OSThreadPriority.Lowest:
+                    return ThreadPriority.Lowest;
+
+                case OSThreadPriority.BelowNormal:
+                    return ThreadPriority.BelowNormal;
+
+                case OSThreadPriority.Normal:
+                    return ThreadPriority.Normal;
+
+                case OSThreadPriority.AboveNormal:
+                    return ThreadPriority.AboveNormal;
+
+                case OSThreadPriority.Highest:
+                case OSThreadPriority.TimeCritical:
+                    return ThreadPriority.Highest;
+
+                case OSThreadPriority.ErrorReturn:
+                    Debug.Fail("GetThreadPriority failed");
+                    return ThreadPriority.Normal;
+
+                default:
+                    return ThreadPriority.Normal;
+            }
+        }
+
+        private static OSThreadPriority MapToOSPriority(ThreadPriority priority)
+        {
+            switch (priority)
+            {
+                case ThreadPriority.Lowest:
+                    return OSThreadPriority.Lowest;
+
+                case ThreadPriority.BelowNormal:
+                    return OSThreadPriority.BelowNormal;
+
+                case ThreadPriority.Normal:
+                    return OSThreadPriority.Normal;
+
+                case ThreadPriority.AboveNormal:
+                    return OSThreadPriority.AboveNormal;
+
+                case ThreadPriority.Highest:
+                    return OSThreadPriority.Highest;
+
+                default:
+                    Environment.FailFast("Unreached");
+                    return OSThreadPriority.Normal;
+            }
+        }
+
+        private bool SetPriority(ThreadPriority priority)
+        {
+            if (_osHandle.IsInvalid)
+            {
+                return true;
```

Comment:
Yes, it is intentional.  Added a comment: "We will set the priority (saved in _priority) when we create an OS thread".